syntax = "proto3";

message Field {
  string name = 1;
  bool is_integer = 2;
}

/* The PointGroup message provides a way to select a group of points 
The fields `scope` and `name` serve as the selector, and the `data` field
names the individual data fields in the Point fields
*/
message PointGroup {
  uint32 id = 1; // primary key
  string scope = 2; // first-level name
  string name = 3;   // second-level name
  repeated Field fields = 4;
}

message Point {
  uint32 group_id = 1; // foreign key to a MetaData message
  uint32 seqnum = 2;  // provides an ordering for all points in the group_id 
  repeated int32 idata = 3; // integer data
  repeated float fdata = 4; // floating point data 
}

message Points {
  repeated Point p = 1;
}

